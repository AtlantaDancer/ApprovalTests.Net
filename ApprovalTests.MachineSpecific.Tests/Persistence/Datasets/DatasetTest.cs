using System;
using System.Data;
using System.Reflection;
using ApprovalTests.Namers;
using ApprovalTests.Persistence.DataSets;
using ApprovalTests.RdlcReports;
using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ReportingDemo;

namespace ApprovalTests.MachineSpecific.Tests.Persistence.Datasets
{
    [TestClass]
//    [UseReporter(typeof (AllFailingTestsClipboardReporter))]
    public class DatasetTest
    {
        private const string ReportName = "ReportingDemo.InsultsReport.rdlc";

        [TestMethod]
        public void TestExtrenalImage()
        {
            RdlcApprovals.VerifyReport("ReportingDemo.ExternalImage.rdlc", GetDefaultData());
        }

        [TestMethod]
        public void TestSimpleReportWith1Dataset()
        {
            RdlcApprovals.VerifyReport(ReportName, GetDefaultData());
        }

        [TestMethod]
        public void TestSimpleReportWithDatasetInAssembly()
        {
            RdlcApprovals.VerifyReport(ReportName, "Model", GetDefaultData());
        }

        [TestMethod]
        public void TestReport()
        {
            RdlcApprovals.VerifyReport(ReportName, GetAssembly(), Tuple.Create("Model", GetDefaultData()));
        }

        [TestMethod]
        public void TestReportWithDataPair()
        {
            RdlcApprovals.VerifyReport(ReportName, GetAssembly(), new DataPairs {{"Model", GetDefaultData()}});
        }

        [TestMethod]
        public void TestSimpleReportExplict()
        {
            RdlcApprovals.VerifyReport(ReportName, GetAssembly(), "Model", GetDefaultData());
        }

        [TestMethod]
        public void TestDataSourceNames()
        {
            NamerFactory.Clear();
            var exception =
                ExceptionUtilities.GetException(
                    () => RdlcApprovals.VerifyReport(ReportName, GetAssembly(), "purposelyMisspelt", GetDefaultData()));
            Approvals.Verify(exception.Message);
        }

        [TestInitialize]
        public void NamerSetUp()
        {
            ApprovalResults.UniqueForMachineName();
        }

        private static DataTable GetDefaultData()
        {
            return new InsultsDataSet.InsultsDataTable().AddTestDataRows(1);
        }

        private static Assembly GetAssembly()
        {
            return typeof (InsultsDataSet).Assembly;
        }
    }
}