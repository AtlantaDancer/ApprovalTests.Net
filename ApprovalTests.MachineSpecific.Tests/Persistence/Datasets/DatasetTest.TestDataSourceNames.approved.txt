The Datasource Name 'purposelyMisspelt'
is not a legal match for RdlcTestTarget.InsultsReport.rdlc,
Legal Matches are: [Model]